### PR TITLE
feat(multisig)!: pretty list of pending proposals

### PR DESCRIPTION
**Pull request by s8sato**
## Context

As of #5217 multisig has some defects especially around the recursion feature:

1. The output of `multisig list` command is verbose, not well structured, and less informative about the authentication status:

    <details>
    <summary>Before</summary>

    ```txt
    "ed01201F89368A4F322263C6F1AEF156759A83FB1AD7D93BAA66BFDFA973ACBADA462F@wonderland"
    "1303a3f8222d9d5d687b8c787bdf6a90a7f774e54a8a37e52fb1681f69e05ba9"
    {
      "approvals": [
        "ed0120987EE8092B2CE4622B4F66D6FE87F5D61575F0D0DFCB2D6B2E8905FE68F685B6@wonderland"
      ],
      "expires_at_ms": 1731520010341,
      "instructions": [
        {
          "SetKeyValue": {
            "Account": {
              "key": "success_marker",
              "object": "ed01201F89368A4F322263C6F1AEF156759A83FB1AD7D93BAA66BFDFA973ACBADA462F@wonderland",
              "value": "congratulations"
            }
          }
        }
      ],
      "proposed_at_ms": 1731516410341
    }
    "ed0120601BBDAF858E77B91494885D0A7618A5E21D24D7E4A672AA4C5C0869D2DDE26F@wonderland"
    "26cec43d4fcfcf3341cfd0c9ef2ceda73a64e7c8dd470eb5268bf33be7a80a51"
    {
      "approvals": [
        "ed01201F89368A4F322263C6F1AEF156759A83FB1AD7D93BAA66BFDFA973ACBADA462F@wonderland"
      ],
      "expires_at_ms": 1731520010341,
      "instructions": [
        {
          "Custom": {
            "payload": {
              "Approve": {
                "account": "ed01201F89368A4F322263C6F1AEF156759A83FB1AD7D93BAA66BFDFA973ACBADA462F@wonderland",
                "instructions_hash": "1303A3F8222D9D5D687B8C787BDF6A90A7F774E54A8A37E52FB1681F69E05BA9"
              }
            }
          }
        }
      ],
      "proposed_at_ms": 1731516410341
    }
    "ed0120D94ED925A4049D78A9F1E55FBF0783A8FE21F2F65CCFE714BBE867A18CC630F8@wonderland"
    "6851493706f2010c6054d7ddf524efcc75a49a5b05bc3e7fdd6a2aea38aa982b"
    {
      "approvals": [
        "ed0120601BBDAF858E77B91494885D0A7618A5E21D24D7E4A672AA4C5C0869D2DDE26F@wonderland"
      ],
      "expires_at_ms": 1731520010341,
      "instructions": [
        {
          "Custom": {
            "payload": {
              "Approve": {
                "account": "ed0120601BBDAF858E77B91494885D0A7618A5E21D24D7E4A672AA4C5C0869D2DDE26F@wonderland",
                "instructions_hash": "26CEC43D4FCFCF3341CFD0C9EF2CEDA73A64E7C8DD470EB5268BF33BE7A80A51"
              }
            }
          }
        }
      ],
      "proposed_at_ms": 1731516410341
    }
    
    ```

    </details>

2. Once a direct multisig account has passed approval, the whole status no longer appears to the child signatories
3. TTL is bound to each account, which is a blocker to consistency as the entire multisig transaction hierarchy
4. On expiration, only the approving node that found the proposal expired is deleted. On execution, only nodes on the path from the last approval to the root proposal are deleted. The remaining nodes require other approvals to be completely deleted
5. Execution of multisig instructions is panic-prone when metadata is modified. Related to #5229

- Hopefully completes #4930

### Solution

1. __feat(multisig): pretty list of pending proposals__

    <details>
    <summary>After</summary>

    ```json
    {
      "6851493706F2010C6054D7DDF524EFCC75A49A5B05BC3E7FDD6A2AEA38AA982B": {
        "instructions": [
          {
            "SetKeyValue": {
              "Account": {
                "object": "ed01201F89368A4F322263C6F1AEF156759A83FB1AD7D93BAA66BFDFA973ACBADA462F@wonderland",
                "key": "success_marker",
                "value": "congratulations"
              }
            }
          }
        ],
        "proposed_at": "2024-11-13T06:50:22Z",
        "expires_in": "59m 59s",
        "approval_path": [
          "1 -> [0/1] ed0120D94ED925A4049D78A9F1E55FBF0783A8FE21F2F65CCFE714BBE867A18CC630F8@wonderland",
          "1 -> [0/1] ed0120601BBDAF858E77B91494885D0A7618A5E21D24D7E4A672AA4C5C0869D2DDE26F@wonderland",
          "1 -> [1/2] ed01201F89368A4F322263C6F1AEF156759A83FB1AD7D93BAA66BFDFA973ACBADA462F@wonderland"
        ]
      }
    }
    
    ```

    </details>

2. __fix(multisig): keep proposals in list while root is pending__
3. __feat(multisig)!: override TTL per transaction__
4. Delete all relevant entries on expiration and execution:
   - __fix(multisig): prune expired path on approve__
   - fix(multisig): keep proposals in list while root is pending
5. __fix(multisig): question existence and format of metadata__

### Migration Guide

- (api-changes) optional `MultisigPropose.transaction_ttl_ms` to override the account default
- (api-changes) `MultisigSpec` `MultisigProposalValue` metadata values

---

### Review notes

Notice the changes to the tests:

- `multisig::multisig_recursion_*` integration tests
- `scripts/tests/multisig.recursion.sh`, especially what `multisig list` output informs

### Checklist

- [x] I've read [`CONTRIBUTING.md`](../CONTRIBUTING.md).
- [x] I've written tests for the code changes.
- [x] All review comments have been resolved.
- [ ] All CI checks pass.

### CODE REVIEW

File: scripts/tests/multisig.recursion.sh
Comment (s8sato) [inline scripts/tests/multisig.recursion.sh:1]: Reference this diagram https://github.com/hyperledger-iroha/iroha/pull/5027#discussion_r1741722664

File: crates/iroha_cli/src/main.rs [hunk 1]
```diff
@@ -1330,53 +1375,147 @@ mod multisig {
...
 
-            trace_back_from(super_account_id.clone(), client, context)?;
+    #[serde_as]
+    #[derive(Debug, Serialize, Constructor)]
+    struct ProposalStatus {
+        instructions: Vec<InstructionBox>,
+        #[serde_as(as = "DisplayFromStr")]
+        proposed_at: humantime::Timestamp,
+        #[serde_as(as = "DisplayFromStr")]
+        expires_in: humantime::Duration,
```
Comment (DCNick3) [inline crates/iroha_cli/src/main.rs:1389]: What do you think about making this a timestamp too or providing expiration timestamp in addition to `expires_in`?

Storing only the expiration time makes interpretation of this object depend on the current time, which might be not nice for tools developed on top of iroha cli.

**Review by DCNick3 — APPROVED** [review]
Only one minor comment

File: crates/iroha_cli/src/main.rs (hunk 1 above)
Comment (s8sato) [reply crates/iroha_cli/src/main.rs:1389]: My assumption was there would be no tools on top of this. Related to https://github.com/hyperledger-iroha/iroha/pull/5240#discussion_r1851685359

Comment (DCNick3) [reply crates/iroha_cli/src/main.rs:1389]: Okay, then it doesn't matter

**Review by mversic — APPROVED** [review]

**Review by mversic — APPROVED** [review]

**Review by DCNick3 — APPROVED** [review]

File: crates/iroha_executor_data_model/src/isi.rs [hunk 2]
```diff
@@ -115,4 +114,64 @@ pub mod multisig {
...
+        /// Proposal contents
+        pub instructions: Vec<InstructionBox>,
+        /// Time in milliseconds at which the proposal was made
+        pub proposed_at_ms: NonZeroU64,
+        /// Time in milliseconds at which the proposal will expire
+        pub expires_at_ms: NonZeroU64,
+        /// List of approvers of the proposal so far
+        pub approvals: BTreeSet<AccountId>,
+        /// In case this proposal is some relaying approval, indicates if it has executed or not
+        pub is_relayed: Option<bool>,
```
Comment (mversic) [inline crates/iroha_executor_data_model/src/isi.rs:141]: @s8sato was it necessary to make it `Option`al?

**Pull request merged**